test/test_downstream.py

```diff
@@ -54,6 +55,36 @@ def test_downstream_move(git_gecko, git_wpt, pull_request, set_pr_status,
     sync = set_pr_status(pr, "success")
     assert sync.gecko_commits[-1].metadata["wpt-type"] == "metadata"
 
+
+def test_wpt_pr_approved(git_gecko, git_wpt, pull_request, set_pr_status,
+                         hg_gecko_try, mock_wpt):
+    mock_wpt.set_data("tests-affected", "")
+
+    pr = pull_request([("Test commit", {"README": "Example change\n"})],
+                      "Test PR")
+    pr._approved = False
+    downstream.new_wpt_pr(git_gecko, git_wpt, pr)
+    sync = set_pr_status(pr, "success")
+
+    sync.data["affected-tests"] = ["example"]
+
+    try_push = sync.latest_try_push
+    assert sync.last_pr_check == {"state": "success", "sha": pr.head}
+    try_push.success = lambda: True
+    with patch('sync.trypush.taskcluster.get_wpt_tasks',
+               return_value=([], [])):
+        downstream.try_push_complete(git_gecko, git_wpt, try_push, sync)
+    assert try_push.status == "complete"
+    assert sync.latest_try_push == try_push
```

Comment:
If I understand right, this is doing the right thing: check that a new (stability) try push hasn't been made yet, despite the previous try push being done.